### PR TITLE
[MFTF] Refactoring of AddOutOfStockProductToCompareListTest

### LINKED ISSUE #31131: [Issue] [MFTF] Refactoring of AddOutOfStockProductToCompareListTest
This issue is automatically created based on existing pull request: magento/magento2#31099: [MFTF] Refactoring of AddOutOfStockProductToCompareListTest

---------
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Create an out of stock product
2. Disable the "Display Out of Stock Products" option
3. Open the PDP 
4. Check if the "Add to compare" link isn't available
5. Enable the "Display Out of Stock Products" option
6. Open the PDP
7. Check if the "Add to compare" link is available
8. Add product to the Comparison list
9. Go to the Comparison list and check if the product is displayed there
10. Clear the Comparison list
11. Go to the product's category
12. Hover the product
13. Add product to the Comparison list
14. Go to  theComparison list and check if the product is displayed there

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Create an out of stock product
2. Disable the "Display Out of Stock Products" option
3. Open the PDP 
4. Check if the "Add to compare" link isn't available
5. Enable the "Display Out of Stock Products" option
6. Open the PDP
7. Check if the "Add to compare" link is available
8. Add product to the Comparison list
9. Go to the Comparison list and check if the product is displayed there
10. Clear the Comparison list
11. Go to the product's category
12. Hover the product
13. Add product to the Comparison list
14. Go to  theComparison list and check if the product is displayed there

### Resolved issues:
1. [x] resolves magento/magento2#31131: [MFTF] Refactoring of AddOutOfStockProductToCompareListTest

### CODE REVIEW

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Catalog/Test/Mftf/Test/AddOutOfStockProductToCompareListTest.xml [hunk 1]
```diff
@@ -30,81 +29,99 @@
                 <requiredEntity createDataKey="category"/>
             </createData>
         </before>
+
         <after>
-            <magentoCLI command="config:set {{CatalogInventoryOptionsShowOutOfStockDisable.path}} {{CatalogInventoryOptionsShowOutOfStockDisable.value}}" stepKey="setConfigShowOutOfStockFalse"/>
+           <magentoCLI command="config:set {{CatalogInventoryOptionsShowOutOfStockDisable.path}} {{CatalogInventoryOptionsShowOutOfStockDisable.value}}" stepKey="setConfigShowOutOfStockFalse"/>
```
Comment (rogyar) [inline app/code/Magento/Catalog/Test/Mftf/Test/AddOutOfStockProductToCompareListTest.xml]: Please, check the trailing spaces at the end

Comment (rogyar) [reply app/code/Magento/Catalog/Test/Mftf/Test/AddOutOfStockProductToCompareListTest.xml]: Also, please, make sure we have a correct indentation for this line of code.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontHoverProductOnCategoryPageActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontHoverProductOnCategoryPageActionGroup">
+        <annotations>
+            <description>Hover product on the Category page</description>
+        </annotations>
+
+    <moveMouseOver selector="{{StorefrontCategoryMainSection.ProductItemInfo}}" stepKey="hoverOverProduct"/>
```
Comment (rogyar) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontHoverProductOnCategoryPageActionGroup.xml]: Please, make sure we have a correct indentation here and everywhere throughout the code.

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @AnnaAPak. Looks good. There are some minor suggestions. Please, check them out. 

Thanks

File: app/code/Magento/Catalog/Test/Mftf/Test/AddOutOfStockProductToCompareListTest.xml (hunk 1 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/Test/AddOutOfStockProductToCompareListTest.xml]: Corrected, thank you.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontHoverProductOnCategoryPageActionGroup.xml (hunk 2 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontHoverProductOnCategoryPageActionGroup.xml]: Corrected and re-checked throughout the code. Thanks.

**Comment by sidolov** [comment]
@magento create issue

**Comment by rogyar** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-8547](https://jira.corp.magento.com/browse/ENGCOM-8547) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
QA not applicable for MFTF Refactoring PR-s. Moving it to _Extended Testing_ column

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Pull request merged**